src/components/Asset/AssetActions/Download.tsx

```diff
@@ -98,7 +104,9 @@ export default function Download({
 
     const isDisabled =
       !asset?.accessDetails.isPurchasable ||
+      !isAssetNetwork ||
```

Comment:
this is such an obvious change and I we just missed that cause it's already in below crazy logic line. Which makes we wonder if we should rather split this up, as on quick look like user could have case where only `! isBalanceSufficient` happens which then would be also missed by below line. Can't remember what exactly the purpose of this line is so least we should do adding code comment describing intention.

Reply:
tested all possible combinations, didn't encounter errors. I have left a comment on it, but I'm not sure why [these conditions](https://github.com/oceanprotocol/market/blob/fix/issue-1356-loader-stuck-button-asset/src/components/Asset/AssetActions/Download.tsx#L108) are grouped though.

Reply:
awesome, thanks for testing and those code comments are really great. We better not touch the line then if all works 🙃